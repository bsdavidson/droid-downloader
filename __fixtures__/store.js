import {Device, DirLine} from "../src/adb";

export const device = "ffffbeefface";

export const deviceFiles = [
  new DirLine(
    "directory",
    "2018-01-02 09:45",
    "folder_1",
    "drwxr-xr-x",
    "3001"
  ),
  new DirLine("file", "2017-09-20 21:22", "file_1", "-rw-r--r--", "42100"),
  new DirLine("file", "2017-10-17 01:45", "file_2", "-rw-r--r--", "3125995")
];

export const deviceFile = deviceFiles[1];

export const devicePath = ["storage", "emulated", "0", "DCIM"];

export const devices = [
  new Device("ffff1337face", "device", {
    model: "SM_G955U"
  }),
  new Device("ffffbeefface", "device", {
    model: "SM_G999V"
  }),
  new Device("ffffbadbeef", "device", {
    model: "SM_G910V"
  })
];

export const localPath = "/tmp";
