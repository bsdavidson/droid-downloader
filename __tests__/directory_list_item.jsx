import React from "react";
import {shallow} from "enzyme";
import renderer from "react-test-renderer";

import DirectoryListItem from "../src/directory_list_item";
import {deviceFiles} from "../__fixtures__/store";

test("<DirectoryListItem /> should render a file", () => {
  const wrapper = renderer
    .create(<DirectoryListItem onClick={() => {}} item={deviceFiles[1]} />)
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});

test("<DirectoryListItem /> should render a directory", () => {
  const wrapper = renderer
    .create(<DirectoryListItem onClick={() => {}} item={deviceFiles[0]} />)
    .toJSON();
  expect(wrapper).toMatchSnapshot();
});

test("<DirectoryListItem /> should call onClick", () => {
  const onClick = jest.fn();
  const wrapper = shallow(
    <DirectoryListItem onClick={onClick} item={deviceFiles[0]} />
  );
  expect(onClick.mock.calls.length).toBe(0);
  wrapper.find(".directory-list-item-link").simulate("click");
  expect(onClick.mock.calls.length).toBe(1);
  expect(onClick.mock.calls[0]).toEqual([deviceFiles[0]]);
});
