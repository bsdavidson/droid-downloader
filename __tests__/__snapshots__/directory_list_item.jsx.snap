// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DirectoryListItem /> should render a directory 1`] = `
<li
  className="directory-list-item directory-list-item-directory"
>
  <a
    className="directory-list-item-link"
    href="#directory"
    onClick={[Function]}
  >
    <i
      className="directory-list-item-icon far fa-folder"
    />
    <span
      className="directory-list-item-elem directory-list-item-name"
    >
      folder_1
    </span>
  </a>
</li>
`;

exports[`<DirectoryListItem /> should render a file 1`] = `
<li
  className="directory-list-item directory-list-item-file"
>
  <a
    className="directory-list-item-link"
    href="#directory"
    onClick={[Function]}
  >
    <i
      className="directory-list-item-icon far fa-file"
    />
    <span
      className="directory-list-item-elem directory-list-item-name"
    >
      file_1
    </span>
  </a>
</li>
`;
